optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -324,12 +336,20 @@ public ScoreDefinition buildDeprecatedScoreDefinition() {
                 throw new IllegalArgumentException("The scoreDirectorFactory cannot have "
                         + "both an easyScoreDirectorFactory and an droolsScoreDirectorFactory.");
             }
+            if (isGenerateDroolsTestOnError()) {
```

Comment:
do not use getter, so it doesn't need to replace null by the default value.

I'd use BooleanUtils.isTrue(generateDroolsTestOnError)